src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -157,7 +163,299 @@ public SPDX30Parser(
         return null;
     }
 
-    public Spdx22Metadata GetMetadata()
+    private void ValidateContext(ContextsResult result)
+    {
+        if (result.Contexts == null || !result.Contexts.Any() || result.Contexts.Count() > 1)
+        {
+            throw new ParserException($"The context property is either empty or has more than one string.");
+        }
+    }
+
+    /// <summary>
+    /// Converts JSON objects to SPDX elements.
+    /// </summary>
+    /// <param name="jsonList"></param>
+    /// <returns></returns>
+    public ElementsResult ConvertToElements(List<object>? jsonList, ref ParserStateResult? result, string? requiredComplianceStandard, IReadOnlyCollection<string>? entitiesWithDifferentNTIARequirements)
+    {
+        var elementsResult = new ElementsResult(result);
+        var elementsList = new List<Element>();
+        var elementsSpdxIdList = new HashSet<string>();
+        var filesList = new List<Parsers.Spdx30SbomParser.Entities.File>();
+
+        if (jsonList is null)
+        {
+            elementsResult.FilesCount = 0;
+            elementsResult.PackagesCount = 0;
+            elementsResult.ReferencesCount = 0;
+            elementsResult.RelationshipsCount = 0;
+            elementsResult.Elements = elementsList;
+            return elementsResult;
+        }
+
+        var complianceStandardAsEnum = ComplianceStandard.None;
+        if (!string.IsNullOrEmpty(this.RequiredComplianceStandard))
+        {
+            if (!Enum.TryParse(this.RequiredComplianceStandard, true, out complianceStandardAsEnum))
+            {
+                throw new ParserException($"{this.RequiredComplianceStandard} compliance standard is not supported.");
+            }
+            else
+            {
+                switch (complianceStandardAsEnum)
+                {
+                    case ComplianceStandard.NTIA:
+                        this.EntitiesToEnforceComplianceStandardsFor = this.entitiesWithDifferentNTIARequirements;
+                        break;
+                    default:
+                        throw new ParserException($"{this.RequiredComplianceStandard} compliance standard is not supported.");
+                }
+            }
+        }
+        else
+        {
+            Console.WriteLine("No required compliance standard.");
+        }
+
+        foreach (JsonObject jsonObject in jsonList)
+        {
```

Comment:
Would the method be easier to read if lines inside the `foreach` loop were in an appropriately named private helper method?

Reply:
Looks like we may have lost this change in the rebasing